### PR TITLE
Add description list basic demo

### PR DESCRIPTION
**Pull request by ajaypratap003**
**What**: Closes https://github.com/patternfly/patternfly-react/issues/7000
Add description list basic demo

**Additional issues**:

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,3 +31,4 @@ lerna-debug.log
 .vscode
 # For vim
 *.swp
+.yarn
```
Comment (kmcfaul) [inline .gitignore:34]: Why did the .gitignore get updated?

Also to fix the DashboardWrapper not being found you will need to also make the change from this PR https://github.com/patternfly/patternfly-react/pull/10191.

Comment (ajaypratap003) [reply .gitignore:34]: @kmcfaul I've updated `.gitignore` due to yarn generate .yarn folder on my machine and want to ignore that folder changes.  Let me know if this change cause any issue 

I've fixed DashboardWrapper not found issue.

**Review by wise-king-sullyman — COMMENTED** [review]
It looks like the content here is pretty different from the content of [the core demo](https://www.patternfly.org/components/description-list/html-demos/basic/), is that something we're concerned with on these @tlabaj ?

File: packages/react-core/src/demos/DescriptionList/DescriptionList.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import BarsIcon from '@patternfly/react-icons/dist/esm/icons/bars-icon';
+import EllipsisVIcon from '@patternfly/react-icons/dist/esm/icons/ellipsis-v-icon';
+import BellIcon from '@patternfly/react-icons/dist/esm/icons/bell-icon';
+import CogIcon from '@patternfly/react-icons/dist/esm/icons/cog-icon';
+import HelpIcon from '@patternfly/react-icons/dist/esm/icons/help-icon';
+import ThIcon from '@patternfly/react-icons/dist/esm/icons/th-icon';
+import QuestionCircleIcon from '@patternfly/react-icons/dist/esm/icons/question-circle-icon';
+import imgAvatar from '@patternfly/react-core/src/components/assets/avatarImg.svg';
+import pfIcon from '../assets/pf-logo-small.svg';
+import pfLogo from '../assets/pf-logo.svg';
```
Comment (thatblindgeye) [inline packages/react-core/src/demos/DescriptionList/DescriptionList.md:15]: Doesn't seem like omitting these imports are necessary as the demo loads fine without them

File: packages/react-core/src/demos/DescriptionList/DescriptionList.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import PlusCircleIcon from '@patternfly/react-icons/dist/esm/icons/plus-circle-icon';
+import { Link } from '@reach/router';
```
Comment (thatblindgeye) [inline packages/react-core/src/demos/DescriptionList/DescriptionList.md:17]: Same as above

File: packages/react-core/src/demos/DescriptionList/examples/DescriptionListBasic.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import { DashboardWrapper } from '@patternfly/react-core/dist/js/demos/DashboardWrapper';
+
+export const DescriptionListBasic: React.FunctionComponent = () => (
+  <DashboardWrapper>
+    <PageSection variant={PageSectionVariants.light}>
+      <Card>
+        <CardBody> Details</CardBody>
+      </Card>
+      <Card>
+        <CardBody>
```
Comment (thatblindgeye) [inline packages/react-core/src/demos/DescriptionList/examples/DescriptionListBasic.tsx:23]: There should only be a single card on the page, with the "Details" text being placed inside a CardHeader using a Title component (as a level h2 heading), and the DescriptionList content placed in the CardBody.

File: packages/react-core/src/demos/DescriptionList/examples/DescriptionListBasic.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  PageSectionVariants,
+  Card,
+  CardBody
+} from '@patternfly/react-core';
+import PlusCircleIcon from '@patternfly/react-icons/dist/esm/icons/plus-circle-icon';
+import { DashboardWrapper } from '@patternfly/react-core/dist/js/demos/DashboardWrapper';
+
+export const DescriptionListBasic: React.FunctionComponent = () => (
+  <DashboardWrapper>
+    <PageSection variant={PageSectionVariants.light}>
```
Comment (thatblindgeye) [inline packages/react-core/src/demos/DescriptionList/examples/DescriptionListBasic.tsx:18]: If we want this demo to be as 1:1 to the Core demo as possible, then this PageSection shouldn't have the variant passed in. There should eb another PageSection before this that contains the "Main title" heading and accompanying additional text "This is a full page demo". The PageSection that contains this "Main title" should have the light variant passed in.

**Review by thatblindgeye — CHANGES_REQUESTED** [review]

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
It looks like there's still some items that aren't lining up with the core demo, for instance the divider not going all the way across the card, the distance between the divider and the heading, the layout and order of the items, and the status being a link:

image

**Comment by kmcfaul** [comment]
Still seeing the divider mismatch on the latest surge.

**Comment by ajaypratap003** [comment]
> Still seeing the divider mismatch on the latest surge.

@kmcfaul @wise-king-sullyman On local changes look fine but seems changes are not reflecting on surge.

image

**Review by wise-king-sullyman — APPROVED** [review]
Yep I see that same thing, not sure why

**Comment by ajaypratap003** [comment]
> Still seeing the divider mismatch on the latest surge.

Seems changes are not reflecting on surge. Please check it on local. It seems fine on local.

**Comment by nicolethoen** [comment]
@mattnolting I think the preview is fixed if you wouldn't mind a quick CSS review

**Review by mattnolting — APPROVED** [review]
Nice, great work! LGTM 🤖

**Pull request merged**